src/actors/objects/mod.rs

```diff
@@ -257,131 +257,80 @@ impl CacheItemRequest for FetchFileDataRequest {
             self.0.file_id.write_sentry_scope(scope);
         });
 
-        let result = request.and_then(move |payload| {
-            if let Some(payload) = payload {
-                let download_dir =
-                    tryf!(path.parent().ok_or(ObjectErrorKind::NoTempDir)).to_owned();
-
-                let download_file = match payload {
-                    DownloadStream::FutureStream(stream) => {
-                        log::trace!("Downloading file for {}", cache_key);
-
-                        let named_download_file = tryf!(
-                            NamedTempFile::new_in(&download_dir).context(ObjectErrorKind::Io)
-                        );
-
-                        let file = tryf!(named_download_file.reopen().context(ObjectErrorKind::Io));
-
-                        let future = stream
-                            .fold(
-                                file,
-                                clone!(threadpool, |mut file, chunk| {
-                                    threadpool
-                                        .spawn_handle(async move {
-                                            file.write_all(&chunk).map(|_| file)
-                                        })
-                                        .boxed_local()
-                                        .compat()
-                                        .map_err(|_| io::Error::from(io::ErrorKind::Other))
-                                        .flatten()
-                                }),
-                            )
-                            .map(move |_| DownloadedFile::Temp(named_download_file));
-
-                        future::Either::A(future)
-                    }
-                    DownloadStream::File(path) => {
-                        log::trace!("Importing downloaded file for {}", cache_key);
-
-                        let future = fs::File::open(&path)
-                            .map(|file| DownloadedFile::Local(path.clone(), file))
-                            .context(ObjectErrorKind::Io)
-                            .map_err(ObjectError::from)
-                            .into_future();
+        let download_dir = tryf!(path.parent().ok_or(ObjectErrorKind::NoTempDir)).to_owned();
+        let download_file =
+            tryf!(NamedTempFile::new_in(&download_dir).context(ObjectErrorKind::Io));
+        let request = download_svc
+            .download(self.0.file_id.clone(), download_file.path().to_owned())
+            .map_err(Into::into);
 
-                        future::Either::B(future)
-                    }
-                };
-
-                let future = download_file.and_then(clone!(threadpool, |download_file| {
+        let result = request.and_then(move |status| -> Result<CacheStatus, ObjectError> {
+            match status {
+                crate::services::download::DownloadStatus::Completed => {
```

Comment:
maybe import this enum? ;-)

Reply:
How strongly do you feel about this?  It's only used in 2 places in this file so is far from a central type used in this file.  Both places have no issues with line splitting/length etc, the code layout would stay the same.  It makes it very obvious that this status lives is part of the download service.  And finally, and a rather weak consideration, while when reading code in your editor this makes only a little difference, it makes a bigger difference when reading code on github where we don't have symbol hints.

I know I'm happier to use longer paths for types then seems to be usual in this code base.  So if there is a desire to keep importing types as much as reasonable then I'm fine to adjust to this.  I don't feel overly strongly, just described my thoughts when choosing the current implementation.

Reply:
I tent to like short names in general. Especially in JS, you *have* to use `import`.

IMO, `std` is the only thing where it makes sense to have 3 segments, like `std::error::Error` or `std::fmt::Display`, etc.

Having 5 segments for a custom type (enum variant in this case) is a bit much. But this is very bikesheddy, and I don’t have strong opinion on this.

Reply:
This is just some bike shedding, so please disregard.

I agree with Arpad here. There are few cases where importing the module is a better choice (I usually go with this for `fmt::Debug` to avoid also importing `fmt::Formatter` and `fmt::Result`), but generally I would only qualify if it is still a short overall name. For example, `std::mem::swap` is still cool. `crate::services::download::DownloadStatus::Completed` is a bit long here.